lib/Mongo/MongoCollection.php

```diff
@@ -96,6 +110,19 @@ public function __get($name)
         return $this->db->selectCollection($this->name . '.' . $name);
     }
 
+    public function __set($name, $value)
+    {
+        if ('w' == $name) {
+            $this->w = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } elseif ('wtimeout' == $name) {
+            $this->wtimeout = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } else {
+            trigger_error('trying to set invalid property', E_ERROR);
```

Comment:
I see your point but `__get` is returning collections so user won't be able to read the value back. Maybe just remove the trigger_error and do nothing.